prime-router/metadata/hl7_mapping/common/observation-result.yml

```diff
@@ -76,11 +87,24 @@ elements:
     schema: ./datatype/sn-structured-numeric
 
   - name: observation-units-identifier
+    condition: >
+      %context.where(extension(%`rsext-units`)).exists() and
+      %context.value.where(extension(%`rsext-cwe-quantity`)).exists().not()
     resource: '%resource.extension(%`rsext-units`).value'
     constants:
       cweFieldPath: '%{hl7OBXField}-6'
     schema: ./datatype/ce-coded-with-exceptions
 
+  - name: observation-units-identifier
+    condition: >
+      %context.value.exists() and
+      %context.value.where(extension(%`rsext-cwe-quantity`)).exists() and
+      %context.value.extension(%`rsext-cwe-quantity`).value.where(extension(%`rsext-hl7v2Name`).value = "units").exists()
+    resource: '%resource.value.extension(%`rsext-cwe-quantity`).value'
+    constants:
+      cweFieldPath: '%{hl7OBXField}-6'
+    schema: ../datatypes/codeableConcept/CWE
```

Comment:
Should there not be a `quantity/CWE` here that takes in `%resource` and the condition checks that `value` is a `Quantity`?

Reply:
The condition check for `value.is(Quantity)` feels like validation so I didn't want to stop a value from being populated based on that condition. I did add it to the OBX-2 mapping in this file to coincide with the existing OBX-2 mapping for "NM" but I could be convinced to leave it or get rid of it

Reply:
so I think that check literally just confirms that the `value` is a `valueQuantity` in the FHIR.  I don't feel super strongly but your advocating for allowing someone to add the `cwe-quantity` extension to a `valueString`?